### ISSUE TITLE
not compatible with pandas >= 2.0.3

### ISSUE THREAD

**Issue by randomrandomguy12345** [rsheftel/pandas_market_calendars#315]
I had hard time finding the list of pandas versions that are compatible with the package. 

Anyway, the package isn't compatible with pandas 2. This is a problem, as pandas 2 did a few changes here and there, and this breaks the code. Downgrading pandas creates a lot of trouble with big projects. Is there a prospect that in near future the package will be compatible with the newer versions of pandas?

**Comment by rsheftel** [comment]
Making it pandas 2.0 compatible is the next item on the task list. For now the latest PyPi release has pandas < 2.0 as a requirement

**Comment by randomrandomguy12345** [comment]
Thanks!

**Comment by LongSimple** [comment]
Thank you

**Comment by programmingwithalex** [comment]
@rsheftel is there any update on when the update will be deployed?

**Comment by nathanramoscfa** [comment]
I forked the project, made this change to allow pandas greater than 2.0:

https://github.com/nathanramoscfa/pandas_market_calendars/commit/0c15491795280123a0b965fb9f0a8b9d65511075

I saw that there were some commits to this project to support pandas 2.1.x so I am trusting that when I install this forked version of the project, it'll work with pandas>=2.0. Well, I installed this forked version to my environment, and reran my Python application, and it worked with no problems. For reference, the pandas compatibility issue came up for me when I tried to build the docs for my application on ReadTheDocs, which would fail, because of this issue.

**Comment by krazykoder** [comment]
@nathanramoscfa  - I am a noob. Appreciate any input. 
Can you write a few lines of instruction on how to build the wheel and install with pip after I fork and git clone this repo. I am having the same issue - some of my code required pandas >2.0.

**Comment by nathanramoscfa** [comment]
> @nathanramoscfa  - I am a noob. Appreciate any input. 
> 
> Can you write a few lines of instruction on how to build the wheel and install with pip after I git clone your fork. I am having the same issue - some of my code required pandas >2.0. 

You would clone my forked version like this to wherever you store your Python projects.

Starting in command prompt, activate your Python environment. I like to use anaconda, so for me it's like this (replace <environment name> with the actual name of your environment):

```
conda activate <environment name>
```

Navigate to your folder where you store all your Python projects, could be something like this:

```
cd C:/Python Projects/
```

Clone my forked version there:

```
git clone https://github.com/nathanramoscfa/pandas_market_calendars.git
```

Then navigate to the project's root directory: 

```
cd pandas_market_calendars
```

Then install as a package to your environment. 

```
pip install .
```

It should then install the forked project you just cloned to your Python environment and now when you import `pandas_market_calendars` into your application, it'll be using my forked version. 

Whenever the `pandas_market_calendars` project updates the PyPi package with the changes we're discussing here, you would then use this command. 

```
pip install --upgrade pandas_market_calendars
```

I just simply needed a solution now, couldn't wait for the devs to update the office PyPi package. But when they do I'll revert back to that version instead of my forked version.

**Comment by rundef** [comment]
FWIW I'm using a slightly older version (4.3.1) with pandas 2.1 and haven't encountered any issue

**Comment by rsheftel** [comment]
Thanks. This is going to require some investigation as this library uses a lot of the deep internals of pandas to work and much of the pandas holiday functionality is undocumented.

**Comment by kentonself** [comment]
Hi.

I am running pandas_market_calendars 4.3.3 with pandas 2.2.0. Everything runs fine, except, I get the following message when I call 

`get_calendar("NYSE").open_at_Time(schedule=x,` timestamp=y)`

> FutureWarning: Downcasting behavior in `replace` is deprecated and will be removed in a future version. To retain the old behavior, explicitly call `result.infer_objects(copy=False)`. To opt-in to the future behavior, set `pd.set_option('future.no_silent_downcasting', True)`

I am assuming this is related to this issue/project.

**Comment by rsheftel** [comment]
v.4.4.0 is published to PyPi and works with Python 3.12 and Pandas 2.2.0
